fakebook/src/app/component/main-view/main-view.component.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { flatten } from '@angular/compiler';
+import { ComponentFixture, fakeAsync, inject, TestBed } from '@angular/core/testing';
+import { By } from '@angular/platform-browser';
+import { NavigationBehaviorOptions , Router } from '@angular/router';
+import { NEVER } from 'rxjs';
+import { AuthService } from '../../service/auth.service';
+import { MainViewComponent } from './main-view.component';
+
+describe('MainViewComponent', () => {
+  let component: MainViewComponent;
+
+  let fixture: ComponentFixture<MainViewComponent>;
+  let navigateSpy: any;
+
+  const FakeAuthService = {
+    isAuth: false,
+    get isAuthenticated(): boolean {
+      return this.isAuth;
+    }
+  };
+
+  beforeEach(async () => {
+
+    const FakeRouterService = {
+      navigateByUrl(url: string, extras?: NavigationBehaviorOptions): Promise<boolean> { return Promise.resolve(true); }
+    };
+
+    await TestBed.configureTestingModule({
+      declarations: [ MainViewComponent ],
+      providers: [
+        { provide: AuthService, useValue: FakeAuthService},
+        { provide: Router, useValue: FakeRouterService },
+      ]
+    })
+    .compileComponents();
+
+    fixture = TestBed.createComponent(MainViewComponent);
+    fixture.detectChanges();
+
+    component = new MainViewComponent(TestBed.inject(AuthService), TestBed.inject(Router));
+    navigateSpy = spyOn(FakeRouterService, 'navigateByUrl');
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should not route if user is not logged in (default value)', () => {
+    const spy = spyOnProperty(FakeAuthService, 'isAuthenticated', 'get');
+
+    component.checkAuthentication();
+
+    expect(navigateSpy).toHaveBeenCalledTimes(0);
+  });
+
+  it('should route if user is logged in', () => {
+    const authSpy = spyOnProperty(FakeAuthService, 'isAuthenticated', 'get').and.returnValue(true);
```

Comment:
clumsy but i guess it works.

Reply:
I did have a feeling this is not the best way to change the value, but I was not sure and when I consulted my team they had no better ideas of how to implement a value to test as true in one spec and false in another